op/model.py

```diff
@@ -15,7 +15,7 @@ def __init__(self, local_unit_name, meta, backend):
         self.app = self.unit.app
         self.relations = RelationMapping(list(meta.relations), self.unit, self._backend, self._cache)
         self.config = ConfigData(self._backend)
-        self.resources = ResourceMapping(list(meta.resources), self._backend)
+        self.resources = {name: Resource(name, backend) for name in meta.resources}
```

Comment:
The previous implementation offered you a bit more freedom to define how to interact with the map of resources. With a dict here, we're pretty much stuck with it to not break people's code, and at least right now it's not entirely clear this is a good way to interact with resources. More details below..